### ISSUE TITLE
No encoders showing up by default

### ISSUE THREAD

**Issue by Techno11** [snobotsim/SnobotSim#13]
Are encoders generated by default or do they have to be added manually?

`ERROR CtreManager:227 - Unknown motor callback: GetQuadraturePosition`

**Comment by pjreiniger** [comment]
Can you include a snippet of your set up and the function call?

**Comment by RobotSpit** [comment]
Thanks,  Don't know exactly what you need to see so Stem to Stern (Two Motors Per Gearbox with a Single Encoder Feedback attached to a Drive Wheel).  We get the same Errors for QuadPosition and QuadVelocity and the Encoder Set-Up does not come up in the Configure Graphics Dashboard either, which makes sense.  Techno11 and I are on the same team, and he must of got busy. :)

	private TalonSRX lF_Drive_1;
	private TalonSRX lF_Drive_2;
	private TalonSRX rF_Drive_1;
	private TalonSRX rF_Drive_2;
	private TalonSRX lR_Drive_1;
	private TalonSRX lR_Drive_2;
	private TalonSRX rR_Drive_1;
	private TalonSRX rR_Drive_2;
.
.
.

    /**
     * Gets the encoder values
     * @return RR, RF, LF, LR Encoder Values
     */
	public double[] getEncoders(){ 
		double[] encoders = {
				rR_Drive_1.getSensorCollection().getQuadraturePosition(),     // RtRear Encoder Wiring Reversed but Motors Reversed So All Good
				-(rF_Drive_2.getSensorCollection().getQuadraturePosition()),  // Motors Reversed 
				-(lF_Drive_2.getSensorCollection().getQuadraturePosition()),  // Motors Reversed 
			        -(lR_Drive_1.getSensorCollection().getQuadraturePosition())   // Motors Reversed 
				};
		return encoders;
	}

	
	public double[] getEncodersVelocity(){ 
		double[] encoders = {
				rR_Drive_1.getSensorCollection().getQuadratureVelocity(),     // RtRear Encoder Wiring Reversed but Motors Reversed So All Good
				-(rF_Drive_2.getSensorCollection().getQuadratureVelocity()),  // Motors Reversed 
				-(lF_Drive_2.getSensorCollection().getQuadratureVelocity()),  // Motors Reversed 
				-(lR_Drive_1.getSensorCollection().getQuadratureVelocity())   // Motors Reversed 
				};
		return encoders;
	}

.
.
.

  /** Method: updateSensorData - Updates the Derived Input Sensor Data.  */
  public void updateSensorData() {
	int idx;
	double tempCnt[] = new double[4];      // (counts)
	double tempRPM[] = new double[4];      // (counts/100msec)
    
    /* Drive Speed Inputs */    
	tempCnt = Robot.kDRIVE.getEncoders();          // counts
	tempRPM = Robot.kDRIVE.getEncodersVelocity();  // counts/100msec

	for (idx=0; idx<4; idx++)
      {
      EncdrCnt[idx] = Math.abs(tempCnt[idx]);
      EncdrVelRaw[idx] = Math.abs(tempRPM[idx]); 
      EncdrRPM[idx] = (EncdrVelRaw[idx]/K_SensorCal.KWSS_Cnt_PulsePerRevEncoder)*(600);  // rpm
      WhlRPM[idx] = EncdrRPM[idx]/K_SensorCal.KWSS_r_EncoderToWheel;                              // rpm
      WhlVel[idx] = (WhlRPM[idx]*K_SensorCal.KWSS_l_DistPerRevWheel)/720;                         // feet/sec
      }
    
    /* UltraSonic Position Inputs */
	USDistRear = calcDistUSRear((float)rearUS.getAverageVoltage());      // inches
	USDistLeft = calcDistUSLeft((float)leftUS.getAverageVoltage());          // inches
	USDistRight = calcDistUSRight((float)rightUS.getAverageVoltage());  // inches

    }

**Comment by pjreiniger** [comment]
Sorry, I got real busy with work after our season ended.  Try 0.11 and let me know if it is better, I wouldn't be surprised if units are wonkey, I did this quick.

**Comment by RobotSpit** [comment]
Thanks.  I will pull it down and give it a try.

**Comment by RobotSpit** [comment]
Trying to install 0.11 (updating from 0.9), I am getting many errors and I an unsuccessful getting the simulation to install properly and run using the posted instructions.  When trying to "install existing project" I get errors stating that the project can not be installed because parts of the project are already installed, but it did not notify what was installed.

So I removed the old project and files from my eclipse workspaces, really anything that referenced the Maven Project, Snobot, SnobotSim, or Simulation, and reinstallled v0.11 from scratch following the instructions an now I get new errors that I didn't see with v0.9.

![image](https://user-images.githubusercontent.com/35512444/38911355-94b1cd5a-429c-11e8-82a0-cd57f59f837e.png)

Info    C:/projects/snobotsim/snobot_sim_java/cpp/lib/SnobotSimJava/Jni/RegisterCallbacksJni.cpp:32 - Initializing the simulator
00:13:13.108 [main] ERROR com.snobot.simulator.joysticks.JoystickDiscoverer - Unknown joystick name 3Dconnexion KMJ Emulator
00:13:13.113 [RobotThread] FATAL com.snobot.simulator.Simulator - Unexpected exception, shutting down simulator
java.lang.NullPointerException: null
	at com.snobot.simulator.robot_container.JavaRobotContainer.startCompetition(JavaRobotContainer.java:32) ~[snobot_sim_gui-all.jar:?]
	at com.snobot.simulator.Simulator$2.run(Simulator.java:318) [snobot_sim_gui-all.jar:?]
	at java.lang.Thread.run(Thread.java:748) [?:1.8.0_152]

**Comment by RobotSpit** [comment]
Sorry to be a bother on this.  It seems though that I did the same install on v0.9 and v0,11 and v0.9 would run with the encoder errors and v0.11 crashes with the null pointers.  Could it be I removed something required when cleaning house when v0.11 would not install due to the "parts of the project already installed" error?  Is there a piece required other than what is contained in the BoilerPlate zip?
